crawler/crawler-ftp/src/main/java/fr/pilato/elasticsearch/crawler/fs/crawler/ftp/FileAbstractorFTP.java

```diff
@@ -0,0 +1,215 @@
+        String fullPath = file.getFullpath();
+        if (isUtf8) {
+            fullPath = new String(fullPath.getBytes(StandardCharsets.UTF_8), FTP.DEFAULT_CONTROL_ENCODING);
+        } else {
+            fullPath = new String(fullPath.getBytes(ALTERNATIVE_ENCODING), FTP.DEFAULT_CONTROL_ENCODING);
+        }
+
+        InputStream inputStream = ftp.retrieveFileStream(fullPath);
+        if (inputStream != null) {
+            return inputStream;
+        } else {
+            throw new IOException(String.format("FTP client can not retrieve stream for [%s]", file.getFullpath()));
+        }
+    }
+
+    @Override
+    public void closeInputStream(InputStream inputStream) throws IOException {
+        inputStream.close();
+        // This is necessary if we want to read the stream
+        ftp.completePendingCommand();
+    }
+
+    @Override
+    public Collection<FileAbstractModel> getFiles(String dir) throws IOException {
+        logger.debug("Listing files from {}", dir);
+        if (isUtf8) {
+            dir = new String(dir.getBytes(StandardCharsets.UTF_8), FTP.DEFAULT_CONTROL_ENCODING);
+        } else {
+            dir = new String(dir.getBytes(ALTERNATIVE_ENCODING), FTP.DEFAULT_CONTROL_ENCODING);
+        }
+
+        FTPFile[] ftpFiles = ftp.listFiles(dir);
+        if (ftpFiles == null) return null;
+        List<FTPFile> files = Arrays.stream(ftpFiles).filter(file -> {
+            if (fsSettings.getFs().isFollowSymlinks()) return true;
+            return !file.isSymbolicLink();
+        }).collect(Collectors.toList());
+
+        Collection<FileAbstractModel> result = new ArrayList<>(files.size());
+        // Iterate other files
+        // We ignore here all files like . and ..
+        String finalDir = dir;
+        result.addAll(files.stream().filter(file -> !".".equals(file.getName()) &&
+                !"..".equals(file.getName()))
+                .map(file -> toFileAbstractModel(finalDir, file))
+                .collect(Collectors.toList()));
+
+        logger.debug("{} local files found", result.size());
+        return result;
+    }
+
+    @Override
+    public boolean exists(String dir) {
+        try {
+            logger.debug("Checking dir existence: " + dir);
+            if (isUtf8) {
+                dir = new String(dir.getBytes(StandardCharsets.UTF_8), FTP.DEFAULT_CONTROL_ENCODING);
+            } else {
+                dir = new String(dir.getBytes(ALTERNATIVE_ENCODING), FTP.DEFAULT_CONTROL_ENCODING);
+            }
+            return ftp.changeWorkingDirectory(dir);
+        } catch (IOException e) {
+            return false;
+        }
+    }
+
+    @Override
+    public void open() throws IOException {
+        Server server = fsSettings.getServer();
+        logger.debug("Opening FTP connection to {}@{}", server.getUsername(), server.getHostname());
+
+        ftp = new FTPClient();
+        ftp.addProtocolCommandListener(ftpListener);
```

Comment:
I believe this is used for debugging, right? 
If so, should we test if we are in the debug level instead of always adding this listener?

Reply:
I can't run docker container with the latest master branch. Could you fix that? I reverted these two commits in our dev branch and it works again. It seems cli is broken by console logger.
https://github.com/dadoonet/fscrawler/commit/f0bc2adce9db25b4d4ca157c8bfc68eef2862d15
https://github.com/dadoonet/fscrawler/commit/b147f17e203436c74dae65251fbaeb696c86e44b

Reply:
Damn! I missed that comment... Need to see what is happening...

Reply:
Some logs for you.
```
/usr/bin/fscrawler: 47: /usr/bin/fscrawler: ps: not found
ERROR StatusLogger Reconfiguration failed: No configuration found for '55054057' at 'null' in 'null'
Exception in thread "main" java.lang.NullPointerException: Cannot invoke "org.apache.logging.log4j.core.appender.ConsoleAppender.addFilter(org.apache.logging.log4j.core.Filter)" because "console" is null
        at fr.pilato.elasticsearch.crawler.fs.cli.FsCrawlerCli.main(FsCrawlerCli.java:132)
```

Reply:
Thanks a ton! #1224 is coming.